ledger/appcow.go

```diff
@@ -425,6 +425,18 @@ func (cb *roundCowState) DelKey(addr basics.Address, aidx basics.AppIndex, globa
 	return nil // note: deletion cannot cause us to violate maxCount
 }
 
+// AppendLog creates a new key-value {addr, aidx} in log
+func (cb *roundCowState) AppendLog(aidx basics.AppIndex, value string) error {
+	// Enforce maximum value length
+	if len(value) > logic.MaxLogSize {
+		return fmt.Errorf("value too long, length was %d", len(value))
+	}
```

Comment:
How is the maximum total length and/or number of logs being checked?

Reply:
it's done in opLog(), https://github.com/shiqizng/go-algorand/blob/opcode/log/data/transactions/logic/eval.go#L3135

Reply:
Oh, right, of course. So why does this code have the MaxLogSize test at all.  Maybe that shouldn't happen at this layer.